packages/frontend/src/components/EditorSettings/Notifications.tsx

```diff
@@ -0,0 +1,146 @@
+import { IFlow } from '@plumber/types'
+
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { useMutation, useQuery } from '@apollo/client'
+import { Flex, Skeleton, Stack, Text } from '@chakra-ui/react'
+import { Menu, useToast } from '@opengovsg/design-system-react'
+import { UPDATE_FLOW_CONFIG } from 'graphql/mutations/update-flow-config'
+import { GET_FLOW } from 'graphql/queries/get-flow'
+
+enum Frequency {
+  Once = 'once',
+  Always = 'always',
+}
+
+const frequencyOptions = [
+  {
+    label: 'First error on this pipe, one email notification per day',
+    value: Frequency.Once,
+  },
+  {
+    label: 'Each error on this pipe',
+    value: Frequency.Always,
+  },
+]
+
+const defaultFrequency = Frequency.Once
+
+export default function Notifications() {
+  const { flowId } = useParams()
+  const [frequency, setFrequency] = useState(defaultFrequency)
+  const { data, loading } = useQuery(GET_FLOW, { variables: { id: flowId } })
+  const flow: IFlow = data?.getFlow
+  const [updateFlowConfig] = useMutation(UPDATE_FLOW_CONFIG)
+  const toast = useToast()
+
+  const onFlowConfigUpdate = useCallback(
+    async (frequency: Frequency) => {
+      await updateFlowConfig({
+        variables: {
+          input: {
+            id: flowId,
+            notificationFrequency: frequency,
+          },
+        },
+        optimisticResponse: {
+          updateFlowConfig: {
+            __typename: 'Flow',
+            id: flow?.id,
+            config: {
+              errorConfig: {
+                notificationFrequency: frequency,
+              },
+            },
+          },
+        },
+      })
+
+      const displayLabel = frequencyOptions
+        .find((option) => option.value === frequency)
+        ?.label.toLowerCase()
+
+      toast({
+        title: `You will now receive an email notification for ${displayLabel}`,
+        status: 'success',
+        duration: 3000,
+        isClosable: true,
+        position: 'top',
+      })
+    },
+    [flow?.id, flowId, updateFlowConfig, toast],
+  )
+
+  const handleClick = useCallback(
+    (frequencyOption: Frequency) => {
+      // only update flow config if a different option is selected
+      if (frequencyOption !== frequency) {
+        setFrequency(frequencyOption)
+        onFlowConfigUpdate(frequencyOption)
+      }
+    },
+    [onFlowConfigUpdate, frequency],
+  )
+
+  useEffect(() => {
+    // Retrieve frequency setting if config was set previously, else set default frequency in config
+    if (!flow) {
+      return
+    }
+    if (flow.config?.errorConfig?.notificationFrequency) {
+      setFrequency(flow.config.errorConfig.notificationFrequency as Frequency)
+    } else {
+      onFlowConfigUpdate(defaultFrequency)
+    }
+  }, [flow, onFlowConfigUpdate])
```

Comment:
comment: The `useEffect` dependency list is strict equality (i.e. reference equality for objects) so this might miss some edge cases (e.g. if someone manually modifies the object)

It's usually safe to reference the full property e.g.
```suggestion
  }, [flow?.config?.errorConfig?.notificationFrequency, onFlowConfigUpdate])
```

Reply:
hmm right now only if i check directly for an empty flow, it won't unnecessarily update the db with the onFlowConfigUpdate. Can I check when can a user manually modify this object on this page?